polymer/src/sample-category-data.html

```diff
@@ -0,0 +1,207 @@
+<!--
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../node_modules/@npm-polymer/polymer/polymer-element.html">
+<link rel="import" href="../node_modules/@npm-polymer/iron-ajax/iron-ajax.html">
+
+<dom-module id="sample-category-data">
+  <template>
+    <iron-ajax auto url="https://ampbyexample.com/sitemap.json" handle-as="json"
+      on-response="handleResponse"></iron-ajax>
+  </template>
+
+  <script>
+
+  (function() {
+
+    class SampleCategoryData extends Polymer.Element {
+
+      static get is() { return 'sample-category-data'; }
+
+      static get properties() { return {
+
+        categoryName: String,
+
+        itemName: String,
+
+        categories: {
+          type: Array,
+          notify: true
+        },
+
+        category: {
+          type: Object,
+          notify: true
+        },
+
+        item: {
+          type: Object,
+          computed: '_computeItem(category.items, itemName)',
+          notify: true
+        },
+
+        failure: {
+          type: Boolean,
+          notify: true,
+          readOnly: true
+        },
+
+        resource: {
+          type: Object
+        }
+
+      }}
+
+      static get observers() { return [
+        '_categoriesLoaded(categories, categoryName)', '_categoryNameSet(categoryName)'
+      ]}
+
+      static get sitemapUrl() { return 'https://ampbyexample.com/sitemap.json'; }
+
+      _categoriesLoaded(categories, categoryName) {
+        this._computeCategory(categoryName)
+      }
+
+      _categoryNameSet(categoryName) {
+        this._fetchCategories(3);
+      }
+
+      handleResponse(data) {
+        this.resource = data.detail.response[0]["categories"]
+        var catNames =  this.resource.map(function(cat) {
+          var obj = {
+            name:  cat["name"],
+            id: cat["id"]
+          };
+          return obj;
+        });
+        this.categories = catNames;
+      }
+
+      _getCategoryObject(categoryName, categories) {
+        if (categories === undefined) {
+          return null;
+        }
+        for (let i = 0, c; c = categories[i]; ++i) {
+          if (c.id === categoryName) {
+            this.category = c;
```

Comment:
this is a bit misleading, the method is called `get` but sets internal state.

Reply:
removed